Assignment10/bugCheck/src/main/java/com/cnu2016/bugCheck/threadBug.java

```
@@ -0,0 +1,60 @@
+package com.cnu2016.bugCheck;
+
+import edu.umd.cs.findbugs.ba.XClass;
+import edu.umd.cs.findbugs.classfile.ClassDescriptor;
+import edu.umd.cs.findbugs.classfile.MethodDescriptor;
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.BytecodeScanningDetector;
+
+import java.util.Arrays;
+
+public class threadBug extends BytecodeScanningDetector {
+
+    private static final boolean DEBUG = Boolean.getBoolean("debug.staticcal");
+
+
+    private BugReporter reporter;
+
+    public threadBug(BugReporter bugReporter) {
+        this.reporter = bugReporter;
+    }
+
+    private String currentClass;
+
+    private boolean check(ClassDescriptor classDescriptor) {
+        if(classDescriptor == null)
+            return false;
+        try {
+            if ("Thread".equals(classDescriptor.getSimpleName()) || check(classDescriptor.getXClass().getSuperclassDescriptor()))
+                return true;
+        } catch(Exception e) {}
```

You are eating exception here. Log it. Also catch specific exceptions.